### PR TITLE
[9.0] knowledge

### PR DESCRIPTION
**Pull request by naousse**
The knowledge module has been deprecated by Odoo SA in version 9.0. This pull request is a port of this module to provide:
- a top-level menu item
- submenus with a settings panel to configure knowledge and document applications
- groups and access rights

### CODE REVIEW

**Comment by pedrobaeza** [comment]
But why do you need this module? This module was only a wrapper to allow certain configuration, but per se doesn't add any model. I don't think it's needed.

**Comment by naousse** [comment]
OK. 
The module document_page need it, and it is the module that add the menu for knowledge operation.

**Comment by naousse** [comment]
So, do i remove it as requirement for document_page module?

**Comment by pedrobaeza** [comment]
Yes, I think it's the way to go. See if in current v9 menu layout, document_page menu entries fit in any place, or create a new root in it. To assure compatibility with another document_\* modules, you can put as XML-ID for this menu "base.menu_document" or similar, and declare the same menu in all modules that need it.

**Comment by naousse** [comment]
OK! Thanks, i will go that way

**Comment by dreispt** [comment]
@pedrobaeza But why create another module adding a new menu root when this module already does that? IMO it's not a problem to keep this, and `document_page`can depend on it for the creation of the top level menu.

**Comment by pedrobaeza** [comment]
No, you don't need another module, Daniel. You only need to declare the menu on subsequent modules. This is the way to implement it: https://github.com/OCA/knowledge/pull/76/files#diff-1b36a8a2d376367ab8f27aac507585a3R5

I also ask if current v9 menu layout  can host menu entries for document_page and the rest of the document_\* modules without creating a parent menu.

**Comment by max3903** [comment]
@pedrobaeza I tend to agree with @dreispt: we should declare the menuitem once in knowledge module instead of every submodules (document_page, document_*)

There is no menu_document in base.

**Comment by max3903** [comment]
@naousse Same remarks as described here: https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-9.0#todo

and add `Odoo Community Association (OCA)` in the author keyword

**Comment by pedrobaeza** [comment]
OK, as you want. I was saying that for avoiding a dumb module. The technique I mention is used on Odoo for Sales, Purchases, and other top menus.

File: knowledge/README.rst [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+Usage
+=====
+
+To use this module, you need to:
+
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/{repo_id}/{branch}
```
Comment (max3903) [inline knowledge/README.rst]: https://runbot.odoo-community.org/runbot/repo/118/9.0

File: knowledge/README.rst [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+.. repo_id is available in https://github.com/OCA/maintainer-tools/blob/master/tools/repos_with_ids.txt
+.. branch is "8.0" for example
```
Comment (max3903) [inline knowledge/README.rst]: These comments can be removed.

File: knowledge/README.rst [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues <https://github.com/OCA/
+{project_repo}/issues>`_.
+In case of trouble, please check there if your issue has already been reported.
+If you spotted it first, help us smashing it by providing a detailed and welcomed feedback `here <https://github.com/OCA/
+{project_repo}/issues/new?body=module:%20
+{module_name}%0Aversion:%20
+{version}%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
```
Comment (max3903) [inline knowledge/README.rst]: Replace {project_repo}, {module_name} and {version}

File: knowledge/README.rst [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+Installation
+============
+
+Makes the Knowledge Application Configuration available from where you can install
+document and Wiki based Hidden.
+
+Configuration
+=============
+
+No configuration required
```
Comment (max3903) [inline knowledge/README.rst]: Go to Knowledge / Configuration / Settings

File: knowledge/README.rst [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+To contribute to this module, please visit http://odoo-community.org.
+
+Changelog
+---------
+
+v9.0.1.0.0
+
+This module is an official module but was not present in odoo 9.0 official repository
+so we took it from Odoo 8.0 repository and add it here.
+
```
Comment (max3903) [inline knowledge/README.rst]: @naousse This is a sufficient description for the changelog. The rest below can be found in the commits and should be removed. Maybe only mention that you made changes to comply with OCA Conventions.

File: knowledge/views/res_config.xml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+                </form>
+            </field>
+        </record>
+        <record id="action_knowledge_configuration" model="ir.actions.act_window">
+            <field name="name">Configure Knowledge</field>
+            <field name="res_model">knowledge.config.settings</field>
+            <field name="view_mode">form</field>
+            <field name="target">inline</field>
+        </record>
+        <menuitem id="menu_knowledge_configuration" name="settings" parent="knowledge.menu_document_configuration"
```
Comment (max3903) [inline knowledge/views/res_config.xml]: name="Settings"

**Comment by naousse** [comment]
I have made the corrections

**Comment by max3903** [comment]
:+1:

File: knowledge/README.rst [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+To use this module, you need to:
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/repo/118/9.0
+
+
+Known issues / Roadmap
+======================
+
```
Comment (dreispt) [inline knowledge/README.rst]: * Add options for other OCA extension modules as they are made available.

**Comment by dreispt** [comment]
Other than a couple of minor notes, it's :+1:

File: knowledge/README.rst [hunk 8]
```diff
@@ -0,0 +1,78 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+   :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+   :alt: License: AGPL-3
+
+================================
+Knowledge Management System base
+================================
+
+This module allows you to write web pages for internal documentation.
```
Comment (max3903) [inline knowledge/README.rst]: This module is the base for any knowledge and document management application.

File: knowledge/README.rst [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+Installation
+============
+
+Makes the Knowledge Application Configuration available from where you can install
+document and Wiki based Hidden.
+
+Configuration
+=============
+
+Go to Knowledge / Configuration / Settings.
```
Comment (max3903) [inline knowledge/README.rst]: `None`

File: knowledge/README.rst [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+
+Usage
+=====
+
+To use this module, you need to:
```
Comment (max3903) [inline knowledge/README.rst]: * go to Knowledge / Configuration / Settings

File: knowledge/README.rst [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+Knowledge Management System base
+================================
+
+This module allows you to write web pages for internal documentation.
+
+Installation
+============
+
+Makes the Knowledge Application Configuration available from where you can install
+document and Wiki based Hidden.
```
Comment (max3903) [inline knowledge/README.rst]: `None`

**Comment by max3903** [comment]
@OCA/tools-maintainers can anyone help us fixing the issue with Travis here?

File: knowledge/__init__.py [hunk 12]
```diff
@@ -0,0 +1,23 @@
+"""Import of model's module."""
```
Comment (max3903) [inline knowledge/__init__.py:1]: @naousse Please remove this line, it makes Travis failed.

File: knowledge/__openerp__.py [hunk 13]
```diff
@@ -0,0 +1,39 @@
+"""Knowledge Management System module."""
```
Comment (max3903) [inline knowledge/__openerp__.py]: @naousse Same here.

File: knowledge/README.rst [hunk 14]
```diff
@@ -0,0 +1,77 @@
...
+
+This module is the base for any knowledge and document management application.
+
+Installation
+============
+
+Configuration
+=============
+
+* Go to Knowledge / Configuration / Settings
```
Comment (max3903) [inline knowledge/README.rst]: `* None`

**Comment by gurneyalex** [comment]
I'm surprised by the WARNINGs in runbot 
https://runbot.odoo-community.org/runbot/build/3119551

File: knowledge/__init__.py (hunk 12 above)
Comment (gurneyalex) [inline knowledge/__init__.py:1]: @max3903 are you sure of this?

File: knowledge/__openerp__.py (hunk 13 above)
Comment (gurneyalex) [inline knowledge/__openerp__.py]: this one probably, linked to this change: 

https://github.com/OCA/maintainer-quality-tools/commit/d8f265b93039a7f0199dcea3a3b92998db62def6

**Comment by gurneyalex** [comment]
https://github.com/OCA/maintainer-quality-tools/issues/295

**Comment by max3903** [comment]
:+1:

File: knowledge/__init__.py (hunk 12 above)
Comment (max3903) [inline knowledge/__init__.py:1]: @gurneyalex Not this one, but in `__openerp__.py`, it does.

**Pull request merged**